src/app/utils/fieldValidator.ts

```diff
@@ -0,0 +1,22 @@
+import { AbstractControl } from '@angular/forms';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FieldValidator {
```

Comment:
As we spoke, I think we should have a RegExFieldValidator that receives a regex as parameter, then maybe two extending clases EmailValidator, PasswordValidator